### PR TITLE
AN-1371_msg sub group and body/auth info to txns

### PR DESCRIPTION
**Pull request by eric-laurello**

### CODE REVIEW

**Comment by desmond-hui** [comment]
@eric-laurello - I believe this transaction should just have a single msg_sub_group but for some reason the last one is a 2nd subgroup

```
select *
from osmosis_dev.silver.msgs
where tx_id = '5862637F9664A034BFF7D0FAEA0B5390D261B292BB65EC211C4F21C51290A35B'
order by msg_group nulls first, msg_sub_group, msg_index;
```

File: models/silver/silver__msgs.sql [hunk 1]
```diff
@@ -25,32 +25,87 @@ WITH b AS (
...
+    WHEN msg_group IS NULL THEN NULL
+    ELSE LAST_VALUE(
+      b.msg_sub_group ignore nulls
+    ) over(
+      PARTITION BY A.tx_id,
+      msg_group
+      ORDER BY
+        A.msg_index DESC rows unbounded preceding
+    )
+  END AS msg_sub_group,
```
Comment (desmond-hui) [inline models/silver/silver__msgs.sql:102]: This should be 0 based to be consistent with our other index columns.  Currently it is 1 based

**Review by desmond-hui — CHANGES_REQUESTED** [review]

**Comment by desmond-hui** [comment]
> @eric-laurello - I believe this transaction should just have a single msg_sub_group but for some reason the last one is a 2nd subgroup
> 
> ```
> select *
> from osmosis_dev.silver.msgs
> where tx_id = '5862637F9664A034BFF7D0FAEA0B5390D261B292BB65EC211C4F21C51290A35B'
> order by msg_group nulls first, msg_sub_group, msg_index;
> ```

This tx `23A20C6A7792F5613B5138FA2F407057223B5F5BEEFA9F82DFD8F69FB1FF7091` has 3 sub groups, should it just be 1?

File: models/silver/silver__msgs.sql (hunk 1 above)
Comment (eric-laurello) [reply models/silver/silver__msgs.sql:102]: adding in a "-1" to the grp cte

**Review by desmond-hui — APPROVED** [review]

**Pull request merged**